network/wsNetwork.go

```diff
@@ -499,13 +517,23 @@ func (wn *WebsocketNetwork) setup() {
 	wn.server.IdleTimeout = httpServerIdleTimeout
 	wn.server.MaxHeaderBytes = httpServerMaxHeaderBytes
 	wn.ctx, wn.ctxCancel = context.WithCancel(context.Background())
-	wn.broadcastQueueHighPrio = make(chan broadcastRequest, 1000)
+	// roughly estimate the number of messages that could be sent over the lifespan of a single round.
+	wn.outgoingMessagesBufferSize = int(config.Consensus[protocol.ConsensusCurrentVersion].NumProposers*2 +
+		config.Consensus[protocol.ConsensusCurrentVersion].SoftCommitteeSize +
+		config.Consensus[protocol.ConsensusCurrentVersion].CertCommitteeSize +
+		config.Consensus[protocol.ConsensusCurrentVersion].NextCommitteeSize +
+		config.Consensus[protocol.ConsensusCurrentVersion].LateCommitteeSize)
```

Comment:
As a small nit, I would change "single round" to "single period" (and say that this is the total number of messages sent at once).

I don't think it makes a big difference here as it's a heuristic, but I would also add `RedoCommitteeSize` and `DownCommitteeSize`. In particular, the committee for the down votes is the largest, at 6000 possible votes. (It needs to be large because it intersects with the cert votes, which are the key committing votes.)

We also pipeline (relaying) all of these votes from the next round and the next period, so it's possible that this number should be 3x as big (as in, we might pipeline 3 periods' worth of votes). On the other hand, this is a pretty unlikely situation and means that the network is experiencing extreme congestion. I think with the current committee sizes, the sum of all our committee sizes is about 20000 messages, which would make 3x about 60000 messages (so with 0.5KB votes this is 30MB).

Reply:
I'll increase the size of the buffer by RedoCommitteeSize+ DownCommitteeSize.